### DISCUSSION TITLE
Store UUIDs as BINARY(16) instead of CHAR(36)

### DISCUSSION THREAD

**Discussion by NicTorgersen** [craftcms/cms#8480] [Ideas]
### Description

CraftCMS could benefit from storing UUIDv4 in BINARY(16) fields in the database instead of todays CHAR(36) according to third party benchmarks.

This blog post illustrates some of the performance differences https://www.stitcher.io/blog/optimised-uuids-in-mysql

A discussion from 2005 about UUIDs on the MySQL forum https://forums.mysql.com/read.php?98,49626,49922#msg-49922

**Comment by angrybrad** [comment]
The UUIDs that are currently in Craft are meant to help us when we get around to content syncing data across environments.  I think they're only in use in once place currently (selecting a user) and that use-case is not frequent.

So open to the idea, but currently not high-priority. Would also have to see if there is something similar for PostgreSQL (which Craft 3 supports) and MSSQL (which would likely be the next database Craft supports). `char`s have the benefit of being lowest common denominator across database platforms.
